### PR TITLE
Upgrade json-schema to 0.4.0 to resolve vulnerability

### LINKED ISSUE #147: `json-schema` is set to a vulnerable version
The solution is upgrading to 0.4.0 and above.

https://github.com/advisories/GHSA-896r-f27r-55mw

### PR DESCRIPTION
**Pull request by markrzen**
Fix #147 

Once upgraded, `npm test` still passes.

https://github.com/kriszyp/json-schema/compare/v0.2.3...v0.4.0

### CODE REVIEW

**Comment by ruiaraujo** [comment]
please update the yarn.lock file by running yarn locally.

**Comment by markrzen** [comment]
Ah, sorry about that, forgot yarn was till being used by folks and totally missed it.

**Comment by markrzen** [comment]
@ruiaraujo Updated - Sorry about the hiccup

File: yarn.lock [hunk 1]
```diff
@@ -3536,11 +3536,16 @@ json-schema-traverse@^0.4.1:
   resolved "https://registry.yarnpkg.com/json-schema-traverse/-/json-schema-traverse-0.4.1.tgz#69f6a87d9513ab8bb8fe63bdb0979c448e684660"
   integrity sha512-xbbCH5dCYU5T8LcEhhuh7HJ88HXuW3qsI3Y0zOZFKfZEHcpWiHU/Jxzk629Brsab/mMiHQti9wMP+845RPe3Vg==
 
-json-schema@0.2.3, json-schema@^0.2.3:
+json-schema@0.2.3:
```
[the remark this answers is not in the document]
Comment (markrzen) [reply yarn.lock:3539]: The dep is super deep in the jest dep chains.

[the remark this answers is not in the document]
Comment (markrzen) [reply yarn.lock:3539]:
```sh
?1 graphql-jit % npm list json-schema
graphql-jit@0.7.1
├─┬ jest@24.7.1
│ └─┬ jest-cli@24.7.1
│   └─┬ jest-config@24.7.1
│     └─┬ jest-environment-jsdom@24.7.1
│       └─┬ jsdom@11.12.0
│         └─┬ request@2.88.0
│           └─┬ http-signature@1.2.0
│             └─┬ jsprim@1.4.1
│               └── json-schema@0.2.3 
└── json-schema@0.4.0 
```

[the remark this answers is not in the document]
Comment (ruiaraujo) [reply yarn.lock:3539]: Yup, I was checking if updating jest is okay.

[the remark this answers is not in the document]
Comment (ruiaraujo) [reply yarn.lock:3539]: It is fine. Please `yarn add --dev ts-jest @types/jest jest` and commit.

[the remark this answers is not in the document]
Comment (ruiaraujo) [reply yarn.lock:3539]: I would also appreciate bumping the version in package.json on the patch given that it is a public dependency.

[the remark this answers is not in the document]
Comment (oporkka) [reply yarn.lock]: I ran now `yarn add --dev ts-jest @types/jest jest` and pushed it

**Comment by ruiaraujo** [comment]
@oporkka I dont know how you can push to this branch, but it seems the jest changes the coverage calculation.

**Comment by oporkka** [comment]
> @oporkka I dont know how you can push to this branch, but it seems the jest changes the coverage calculation.

I was just trying, and probably the "allow maintainers to push" option was selected for the PR.

Here are the coverage results after the jest update
image

Here are the coverage results from the PR where default branch was switched to `main`:
image

Note that branch coverage seems to have gone down but the statement coverage up. Since we did not change code here, just updated libraries, and this made our coverage drop, I propose we react by dropping the branch coverage target in configuration to 92.

**Comment by oporkka** [comment]
:+1:

File: package.json [hunk 2]
```diff
@@ -39,7 +39,7 @@
     },
     "coverageThreshold": {
       "global": {
-        "branches": 94,
+        "branches": 92,
```
Comment (oporkka) [inline package.json:42]: This has been reduced now to react to the reduced branch coverage measured after the update of ts-jest library.

**Comment by ruiaraujo** [comment]
:+1:

**Pull request merged**